lib/src/intl/number_format.dart

```diff
@@ -591,7 +591,12 @@ class NumberFormat {
       // a fixed-size integer type, so we extract each of the three as an
       // integer pieces.
       integerPart = _floor(number);
-      var fraction = number - integerPart;
+      double fraction;
+      if (integerPart != number) {
+        fraction = double.parse('0.${number.toString().split('.').last}');
```

Comment:
What problem is this trying to solve?

Reply:
Please check my issue #337, I have even created a repo to demo the issue.